daemon/smartcard.c

```diff
@@ -79,20 +91,150 @@ bytes_to_string_new(unsigned char *data, size_t len)
 	return str;
 }
 
+static TokenType
+smartcard_tokentype_to_proto(container_token_type_t tokentype)
+{
+	switch (tokentype) {
+	case CONTAINER_TOKEN_TYPE_NONE:
+		return TOKEN_TYPE__NONE;
+	case CONTAINER_TOKEN_TYPE_DEVICE:
+		return TOKEN_TYPE__DEVICE;
+	case CONTAINER_TOKEN_TYPE_USB:
+		return TOKEN_TYPE__USB;
+	default:
+		FATAL("Invalid container_token_type_t value : %d", tokentype);
```

Comment:
taht would break existing code structure (see smartcard_hashalgo_to_proto() or control_container_type_to_proto() as examples). Furthermore, this case must not ever happen and if it does a serious error condition must have been encountered. Probably it is the safest thing to do to stop execution here